src/auth/src/credentials/external_account.rs

```diff
@@ -349,6 +350,159 @@ impl Builder {
+/// ```
+/// # use google_cloud_auth::credentials::external_account::ProgrammaticBuilder;
+/// # use google_cloud_auth::credentials::subject_token::{SubjectTokenProvider, SubjectToken, Builder as SubjectTokenBuilder};
+/// # use google_cloud_auth::errors::SubjectTokenProviderError;
+/// # use std::error::Error;
+/// # use std::fmt;
+/// # use std::sync::Arc;
+/// #
+/// # #[derive(Debug)]
+/// # struct MyTokenProvider;
+/// #
+/// # #[derive(Debug)]
+/// # struct MyProviderError;
+/// # impl fmt::Display for MyProviderError { fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result { write!(f, "MyProviderError") } }
+/// # impl Error for MyProviderError {}
+/// # impl SubjectTokenProviderError for MyProviderError { fn is_transient(&self) -> bool { false } }
+/// #
+/// # #[async_trait::async_trait]
+/// # impl SubjectTokenProvider for MyTokenProvider {
+/// #     type Error = MyProviderError;
+/// #     async fn subject_token(&self) -> Result<SubjectToken, Self::Error> {
+/// #         Ok(SubjectTokenBuilder::new("my-programmatic-token".to_string()).build())
+/// #     }
+/// # }
+/// #
+/// # tokio_test::block_on(async {
+/// let provider = Arc::new(MyTokenProvider);
+///
+/// let credentials = ProgrammaticBuilder::new(provider)
+///     .with_audience("//iam.googleapis.com/projects/123/locations/global/workloadIdentityPools/my-pool/providers/my-provider".to_string())
+///     .with_subject_token_type("urn:ietf:params:oauth:token-type:jwt".to_string())
+///     .with_token_url("https://sts.googleapis.com/v1beta/token".to_string())
+///     .with_quota_project_id("my-quota-project")
+///     .with_scopes(vec!["https://www.googleapis.com/auth/devstorage.read_only".to_string()])
+///     .build()
+///     .unwrap();
+/// # });
+/// ```
+/// [SubjectTokenProvider]: crate::credentials::subject_token::SubjectTokenProvider
+pub struct ProgrammaticBuilder {
+    quota_project_id: Option<String>,
+    subject_token_provider: Arc<dyn dynamic::SubjectTokenProvider>,
+    config: ExternalAccountConfigBuilder,
+}
+
+impl ProgrammaticBuilder {
+    /// Creates a new builder that uses the provided [`SubjectTokenProvider`] to
+    /// fetch the third-party subject token.
+    pub fn new(subject_token_provider: Arc<dyn dynamic::SubjectTokenProvider>) -> Self {
+        Self {
+            subject_token_provider,
+            quota_project_id: None,
+            config: ExternalAccountConfigBuilder::default(),
+        }
+    }
+
+    /// Sets the optional [quota project] for this credentials.
+    ///
+    /// In some services, you can use a service account in
+    /// one project for authentication and authorization, and charge
+    /// the usage to a different project. This requires that the
+    /// service account has `serviceusage.services.use` permissions on the quota project.
+    ///
+    /// [quota project]: https://cloud.google.com/docs/quotas/quota-project
+    pub fn with_quota_project_id<S: Into<String>>(mut self, quota_project_id: S) -> Self {
```

Comment:
missing examples on the builder methods

Reply:
We are adding for all public builder methods, irrespective of the level of difficulty. See 

https://github.com/googleapis/google-cloud-rust/blob/b5c298dd927ff5a20775e5b898f59c394612c51d/src/auth/src/credentials/user_account.rs#L176-L186